src/widgetastic/widget/base.py

```diff
@@ -950,45 +953,28 @@ def fill(self, values):
 
         Args:
             values: A dictionary of ``widget_name: value_to_fill``.
+            wait_widget: wait_for value. If wait > 0 widget.wait_displayed(wait) is called for every
+                         widget
 
         Returns:
             :py:class:`bool` if the fill changed any value.
         """
+        changed = []
         values = deflatten_dict(values)
         self.last_fill_data = values
-        was_change = False
-        b_fill = self.before_fill(values)
-        if b_fill is True:
-            # Only on True, nothing else
-            was_change = True
+        changed.append(self.before_fill(values))
+
         extra_keys = set(values.keys()) - set(self.widget_names)
         if extra_keys:
             self.logger.warning(
                 'Extra values that have no corresponding fill fields passed: %s',
                 ', '.join(extra_keys))
-        for name in self.widget_names:
-            widget = getattr(self, name)
-            if name not in values or values[name] is None:
-                if name not in values:
-                    self.logger.debug(
-                        'Skipping fill of %r because value was not specified', name)
-                else:
-                    self.logger.debug(
-                        'Skipping fill of %r because value was None', name)
-                continue
+        to_fill = [(getattr(self, n), values[n]) for n in self.widget_names
+                   if n in values and values[n] is not None]
```

Comment:
:100:

Reply:
While I love this simplification, I think that its better expressed as a full loop with logging. Otherwise we lose some logging messages here for why a view's widgets were not filled.

Reply:
I just can't recall when we used those debug messages at least once. Do you think those necessary ?